### PR TITLE
[BE] consolidate 4-GPU integration tests into 8-GPU tests and reduce frequency

### PR DESCRIPTION
**Pull request by tianyu-l**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #745
* #741
* #740

Previously CI has two parts of GPU tests: (1) 4 GPU test which runs hourly (2) 8 GPU test which runs daily.
- I think the frequency of (1) is unnecessary, the only thing outside PR changes which can break CI is pytorch-nightly release. This PR reduces the frequency to e.g. per-6-hours (4 times a day).
- Since we reduce the frequency of (1) to be close to (2), can we consolidate them into one yaml. The overall “cost” might be lower instead of higher considering we only launch the container once. Also it’s going to be less confusing if we just show a single badge as “integration tests” on README.

Results:
- CI used to run ~16min on 4-GPU test and ~10min on 8-GPU test
- It now runs 18.5min on the 8-GPU test -- the main saving is from only pulling docker image once (around 4 min)

### CODE REVIEW

File: .github/workflows/integration_test_8gpu.yaml [hunk 1]
```diff
@@ -21,7 +21,7 @@ jobs:
     with:
       runner: linux.g5.48xlarge.nvidia.gpu
       gpu-arch-type: cuda
-      gpu-arch-version: "12.1"
+      gpu-arch-version: "12.4"
```
Comment (XilunWu) [inline .github/workflows/integration_test_8gpu.yaml:24]: Has someone been running local tests with 12.4 CUDA? Does it work?

File: README.md [hunk 2]
```diff
@@ -1,5 +1,4 @@
-[![4 GPU Integration Test](https://github.com/pytorch/torchtitan/actions/workflows/integration_test_4gpu.yaml/badge.svg?branch=main)](https://github.com/pytorch/torchtitan/actions/workflows/integration_test_4gpu.yaml?query=branch%3Amain)
-[![8 GPU Integration Test](https://github.com/pytorch/torchtitan/actions/workflows/integration_test_8gpu.yaml/badge.svg?branch=main)](https://github.com/pytorch/torchtitan/actions/workflows/integration_test_8gpu.yaml?query=branch%3Amain)
+[![Integration Tests](https://github.com/pytorch/torchtitan/actions/workflows/integration_test_8gpu.yaml/badge.svg?branch=main)](https://github.com/pytorch/torchtitan/actions/workflows/integration_test_8gpu.yaml?query=branch%3Amain)
```
Comment (XilunWu) [inline README.md]: Will we only use the 8gpu AWS instance for CI from now on?

**Review by XilunWu — COMMENTED** [review]
Left some questions.

File: .github/workflows/integration_test_8gpu.yaml (hunk 1 above)
Comment (tianyu-l) [reply .github/workflows/integration_test_8gpu.yaml:24]: I haven't. To be honest I'm not sure.
This change is mainly for consistency with https://github.com/pytorch/torchtitan/pull/745/files#diff-e327f3f247423713ee949ef4eef6b82de392abca8c53137159d82f073510c4f9R39

File: README.md (hunk 2 above)
Comment (tianyu-l) [reply README.md]: Yes, this is the idea of this PR. Please share concerns you may have!

**Review by fegin — APPROVED** [review]

**Comment by tianyu-l** [comment]
encounter ghstack issues, reopening in https://github.com/pytorch/torchtitan/pull/750

**Pull request closed without merging**